# XOR Decryption
Source: https://projecteuler.net/problem=59

# Problem Statement:
Each character on a computer is assigned a unique code and the preferred standard is ASCII (American Standard Code for Information Interchange). For example, uppercase A = 65, asterisk (*) = 42, and lowercase k = 107.

A modern encryption method is to take a text file, convert the bytes to ASCII, then XOR each byte with a given value, taken from a secret key. The advantage with the XOR function is that using the same encryption key on the cipher text, restores the plain text; for example, 65 XOR 42 = 107, then 107 XOR 42 = 65.

For unbreakable encryption, the key is the same length as the plain text message, and the key is made up of random bytes. The user would keep the encrypted message and the encryption key in different locations, and without both "halves", it is impossible to decrypt the message.

Unfortunately, this method is impractical for most users, so the modified method is to use a password as a key. If the password is shorter than the message, which is likely, the key is repeated cyclically throughout the message. The balance for this method is using a sufficiently long password key for security, but short enough to be memorable.

Your task has been made easy, as the encryption key consists of three lower case characters. Using p059_cipher.txt (right click and 'Save Link/Target As...'), a file containing the encrypted ASCII codes, and the knowledge that the plain text must contain common English words, decrypt the message and find the sum of the ASCII values in the original text.

# Note:
the ASCII values for lowercase characters starts at a = 97 and ends at z = 122

In [1]:
## testing
chr(97)

'a'

In [3]:
chr(122)

'z'

In [5]:
## xplore file
with open('p059_cipher.txt') as file:
    for idx, line in enumerate(file):
        print(idx)
        print(line)
        

0
36,22,80,0,0,4,23,25,19,17,88,4,4,19,21,11,88,22,23,23,29,69,12,24,0,88,25,11,12,2,10,28,5,6,12,25,10,22,80,10,30,80,10,22,21,69,23,22,69,61,5,9,29,2,66,11,80,8,23,3,17,88,19,0,20,21,7,10,17,17,29,20,69,8,17,21,29,2,22,84,80,71,60,21,69,11,5,8,21,25,22,88,3,0,10,25,0,10,5,8,88,2,0,27,25,21,10,31,6,25,2,16,21,82,69,35,63,11,88,4,13,29,80,22,13,29,22,88,31,3,88,3,0,10,25,0,11,80,10,30,80,23,29,19,12,8,2,10,27,17,9,11,45,95,88,57,69,16,17,19,29,80,23,29,19,0,22,4,9,1,80,3,23,5,11,28,92,69,9,5,12,12,21,69,13,30,0,0,0,0,27,4,0,28,28,28,84,80,4,22,80,0,20,21,2,25,30,17,88,21,29,8,2,0,11,3,12,23,30,69,30,31,23,88,4,13,29,80,0,22,4,12,10,21,69,11,5,8,88,31,3,88,4,13,17,3,69,11,21,23,17,21,22,88,65,69,83,80,84,87,68,69,83,80,84,87,73,69,83,80,84,87,65,83,88,91,69,29,4,6,86,92,69,15,24,12,27,24,69,28,21,21,29,30,1,11,80,10,22,80,17,16,21,69,9,5,4,28,2,4,12,5,23,29,80,10,30,80,17,16,21,69,27,25,23,27,28,0,84,80,22,23,80,17,16,17,17,88,25,3,88,4,13,29,80,17,10,5,0,88,3,16,21,80,10,30,80,17,16,25

In [6]:
## get ASCII values into a list of ints
ASCII_int_list = []

with open('p059_cipher.txt') as file:
    for line in file:
        
        line_list = line.split(',')
        
        for elem in line_list:
            ASCII_int_list.append(int(elem))
            
print(ASCII_int_list)

[36, 22, 80, 0, 0, 4, 23, 25, 19, 17, 88, 4, 4, 19, 21, 11, 88, 22, 23, 23, 29, 69, 12, 24, 0, 88, 25, 11, 12, 2, 10, 28, 5, 6, 12, 25, 10, 22, 80, 10, 30, 80, 10, 22, 21, 69, 23, 22, 69, 61, 5, 9, 29, 2, 66, 11, 80, 8, 23, 3, 17, 88, 19, 0, 20, 21, 7, 10, 17, 17, 29, 20, 69, 8, 17, 21, 29, 2, 22, 84, 80, 71, 60, 21, 69, 11, 5, 8, 21, 25, 22, 88, 3, 0, 10, 25, 0, 10, 5, 8, 88, 2, 0, 27, 25, 21, 10, 31, 6, 25, 2, 16, 21, 82, 69, 35, 63, 11, 88, 4, 13, 29, 80, 22, 13, 29, 22, 88, 31, 3, 88, 3, 0, 10, 25, 0, 11, 80, 10, 30, 80, 23, 29, 19, 12, 8, 2, 10, 27, 17, 9, 11, 45, 95, 88, 57, 69, 16, 17, 19, 29, 80, 23, 29, 19, 0, 22, 4, 9, 1, 80, 3, 23, 5, 11, 28, 92, 69, 9, 5, 12, 12, 21, 69, 13, 30, 0, 0, 0, 0, 27, 4, 0, 28, 28, 28, 84, 80, 4, 22, 80, 0, 20, 21, 2, 25, 30, 17, 88, 21, 29, 8, 2, 0, 11, 3, 12, 23, 30, 69, 30, 31, 23, 88, 4, 13, 29, 80, 0, 22, 4, 12, 10, 21, 69, 11, 5, 8, 88, 31, 3, 88, 4, 13, 17, 3, 69, 11, 21, 23, 17, 21, 22, 88, 65, 69, 83, 80, 84, 87, 68, 69, 83, 80, 84, 87, 7

In [24]:
## testing
sample_message = ASCII_int_list[0:9] + ASCII_int_list[-9:]

for first in range(97, 122 + 1):
    for second in range(97, 122 + 1):
#             second = 97
        for third in range(97, 122 + 1):
#             third = 97
    
            decoded_message = []
    
            decoder = [first, second, third]
            
            for idx in range(len(sample_message)):
                
                decoded_ASCII = sample_message[idx] ^ decoder[idx%3]
                
                decoded_message.append(decoded_ASCII)
                
                
            is_valid=True
            
            if (decoded_message[0] >= 65 and decoded_message[0] <= 90) or (decoded_message[0] >= 97 and decoded_message[0] <= 122):
                decoded_message = [chr(ASCII) for ASCII in decoded_message]
            
                print(decoder, decoded_message)

[97, 97, 97] ['E', 'w', '1', 'a', 'a', 'e', 'v', 'x', 'r', '$', 'w', 'd', 'i', '{', 't', 'v', 'j', '?']
[97, 97, 98] ['E', 'w', '2', 'a', 'a', 'f', 'v', 'x', 'q', '$', 'w', 'g', 'i', '{', 'w', 'v', 'j', '<']
[97, 97, 99] ['E', 'w', '3', 'a', 'a', 'g', 'v', 'x', 'p', '$', 'w', 'f', 'i', '{', 'v', 'v', 'j', '=']
[97, 97, 100] ['E', 'w', '4', 'a', 'a', '`', 'v', 'x', 'w', '$', 'w', 'a', 'i', '{', 'q', 'v', 'j', ':']
[97, 97, 101] ['E', 'w', '5', 'a', 'a', 'a', 'v', 'x', 'v', '$', 'w', '`', 'i', '{', 'p', 'v', 'j', ';']
[97, 97, 102] ['E', 'w', '6', 'a', 'a', 'b', 'v', 'x', 'u', '$', 'w', 'c', 'i', '{', 's', 'v', 'j', '8']
[97, 97, 103] ['E', 'w', '7', 'a', 'a', 'c', 'v', 'x', 't', '$', 'w', 'b', 'i', '{', 'r', 'v', 'j', '9']
[97, 97, 104] ['E', 'w', '8', 'a', 'a', 'l', 'v', 'x', '{', '$', 'w', 'm', 'i', '{', '}', 'v', 'j', '6']
[97, 97, 105] ['E', 'w', '9', 'a', 'a', 'm', 'v', 'x', 'z', '$', 'w', 'l', 'i', '{', '|', 'v', 'j', '7']
[97, 97, 106] ['E', 'w', ':', 'a', 'a', 'n', 'v', 'x', 'y'

[98, 103, 101] ['F', 'q', '5', 'b', 'g', 'a', 'u', '~', 'v', "'", 'q', '`', 'j', '}', 'p', 'u', 'l', ';']
[98, 103, 102] ['F', 'q', '6', 'b', 'g', 'b', 'u', '~', 'u', "'", 'q', 'c', 'j', '}', 's', 'u', 'l', '8']
[98, 103, 103] ['F', 'q', '7', 'b', 'g', 'c', 'u', '~', 't', "'", 'q', 'b', 'j', '}', 'r', 'u', 'l', '9']
[98, 103, 104] ['F', 'q', '8', 'b', 'g', 'l', 'u', '~', '{', "'", 'q', 'm', 'j', '}', '}', 'u', 'l', '6']
[98, 103, 105] ['F', 'q', '9', 'b', 'g', 'm', 'u', '~', 'z', "'", 'q', 'l', 'j', '}', '|', 'u', 'l', '7']
[98, 103, 106] ['F', 'q', ':', 'b', 'g', 'n', 'u', '~', 'y', "'", 'q', 'o', 'j', '}', '\x7f', 'u', 'l', '4']
[98, 103, 107] ['F', 'q', ';', 'b', 'g', 'o', 'u', '~', 'x', "'", 'q', 'n', 'j', '}', '~', 'u', 'l', '5']
[98, 103, 108] ['F', 'q', '<', 'b', 'g', 'h', 'u', '~', '\x7f', "'", 'q', 'i', 'j', '}', 'y', 'u', 'l', '2']
[98, 103, 109] ['F', 'q', '=', 'b', 'g', 'i', 'u', '~', '~', "'", 'q', 'h', 'j', '}', 'x', 'u', 'l', '3']
[98, 103, 110] ['F', 'q', '>', 'b', 'g',

[99, 97, 114] ['G', 'w', '"', 'c', 'a', 'v', 't', 'x', 'a', '&', 'w', 'w', 'k', '{', 'g', 't', 'j', ',']
[99, 97, 115] ['G', 'w', '#', 'c', 'a', 'w', 't', 'x', '`', '&', 'w', 'v', 'k', '{', 'f', 't', 'j', '-']
[99, 97, 116] ['G', 'w', '$', 'c', 'a', 'p', 't', 'x', 'g', '&', 'w', 'q', 'k', '{', 'a', 't', 'j', '*']
[99, 97, 117] ['G', 'w', '%', 'c', 'a', 'q', 't', 'x', 'f', '&', 'w', 'p', 'k', '{', '`', 't', 'j', '+']
[99, 97, 118] ['G', 'w', '&', 'c', 'a', 'r', 't', 'x', 'e', '&', 'w', 's', 'k', '{', 'c', 't', 'j', '(']
[99, 97, 119] ['G', 'w', "'", 'c', 'a', 's', 't', 'x', 'd', '&', 'w', 'r', 'k', '{', 'b', 't', 'j', ')']
[99, 97, 120] ['G', 'w', '(', 'c', 'a', '|', 't', 'x', 'k', '&', 'w', '}', 'k', '{', 'm', 't', 'j', '&']
[99, 97, 121] ['G', 'w', ')', 'c', 'a', '}', 't', 'x', 'j', '&', 'w', '|', 'k', '{', 'l', 't', 'j', "'"]
[99, 97, 122] ['G', 'w', '*', 'c', 'a', '~', 't', 'x', 'i', '&', 'w', '\x7f', 'k', '{', 'o', 't', 'j', '$']
[99, 98, 97] ['G', 't', '1', 'c', 'b', 'e', 't', '{'

[101, 99, 103] ['A', 'u', '7', 'e', 'c', 'c', 'r', 'z', 't', ' ', 'u', 'b', 'm', 'y', 'r', 'r', 'h', '9']
[101, 99, 104] ['A', 'u', '8', 'e', 'c', 'l', 'r', 'z', '{', ' ', 'u', 'm', 'm', 'y', '}', 'r', 'h', '6']
[101, 99, 105] ['A', 'u', '9', 'e', 'c', 'm', 'r', 'z', 'z', ' ', 'u', 'l', 'm', 'y', '|', 'r', 'h', '7']
[101, 99, 106] ['A', 'u', ':', 'e', 'c', 'n', 'r', 'z', 'y', ' ', 'u', 'o', 'm', 'y', '\x7f', 'r', 'h', '4']
[101, 99, 107] ['A', 'u', ';', 'e', 'c', 'o', 'r', 'z', 'x', ' ', 'u', 'n', 'm', 'y', '~', 'r', 'h', '5']
[101, 99, 108] ['A', 'u', '<', 'e', 'c', 'h', 'r', 'z', '\x7f', ' ', 'u', 'i', 'm', 'y', 'y', 'r', 'h', '2']
[101, 99, 109] ['A', 'u', '=', 'e', 'c', 'i', 'r', 'z', '~', ' ', 'u', 'h', 'm', 'y', 'x', 'r', 'h', '3']
[101, 99, 110] ['A', 'u', '>', 'e', 'c', 'j', 'r', 'z', '}', ' ', 'u', 'k', 'm', 'y', '{', 'r', 'h', '0']
[101, 99, 111] ['A', 'u', '?', 'e', 'c', 'k', 'r', 'z', '|', ' ', 'u', 'j', 'm', 'y', 'z', 'r', 'h', '1']
[101, 99, 112] ['A', 'u', ' ', 'e', 'c',

[102, 102, 99] ['B', 'p', '3', 'f', 'f', 'g', 'q', '\x7f', 'p', '#', 'p', 'f', 'n', '|', 'v', 'q', 'm', '=']
[102, 102, 100] ['B', 'p', '4', 'f', 'f', '`', 'q', '\x7f', 'w', '#', 'p', 'a', 'n', '|', 'q', 'q', 'm', ':']
[102, 102, 101] ['B', 'p', '5', 'f', 'f', 'a', 'q', '\x7f', 'v', '#', 'p', '`', 'n', '|', 'p', 'q', 'm', ';']
[102, 102, 102] ['B', 'p', '6', 'f', 'f', 'b', 'q', '\x7f', 'u', '#', 'p', 'c', 'n', '|', 's', 'q', 'm', '8']
[102, 102, 103] ['B', 'p', '7', 'f', 'f', 'c', 'q', '\x7f', 't', '#', 'p', 'b', 'n', '|', 'r', 'q', 'm', '9']
[102, 102, 104] ['B', 'p', '8', 'f', 'f', 'l', 'q', '\x7f', '{', '#', 'p', 'm', 'n', '|', '}', 'q', 'm', '6']
[102, 102, 105] ['B', 'p', '9', 'f', 'f', 'm', 'q', '\x7f', 'z', '#', 'p', 'l', 'n', '|', '|', 'q', 'm', '7']
[102, 102, 106] ['B', 'p', ':', 'f', 'f', 'n', 'q', '\x7f', 'y', '#', 'p', 'o', 'n', '|', '\x7f', 'q', 'm', '4']
[102, 102, 107] ['B', 'p', ';', 'f', 'f', 'o', 'q', '\x7f', 'x', '#', 'p', 'n', 'n', '|', '~', 'q', 'm', '5']
[102, 10

[102, 121, 104] ['B', 'o', '8', 'f', 'y', 'l', 'q', '`', '{', '#', 'o', 'm', 'n', 'c', '}', 'q', 'r', '6']
[102, 121, 105] ['B', 'o', '9', 'f', 'y', 'm', 'q', '`', 'z', '#', 'o', 'l', 'n', 'c', '|', 'q', 'r', '7']
[102, 121, 106] ['B', 'o', ':', 'f', 'y', 'n', 'q', '`', 'y', '#', 'o', 'o', 'n', 'c', '\x7f', 'q', 'r', '4']
[102, 121, 107] ['B', 'o', ';', 'f', 'y', 'o', 'q', '`', 'x', '#', 'o', 'n', 'n', 'c', '~', 'q', 'r', '5']
[102, 121, 108] ['B', 'o', '<', 'f', 'y', 'h', 'q', '`', '\x7f', '#', 'o', 'i', 'n', 'c', 'y', 'q', 'r', '2']
[102, 121, 109] ['B', 'o', '=', 'f', 'y', 'i', 'q', '`', '~', '#', 'o', 'h', 'n', 'c', 'x', 'q', 'r', '3']
[102, 121, 110] ['B', 'o', '>', 'f', 'y', 'j', 'q', '`', '}', '#', 'o', 'k', 'n', 'c', '{', 'q', 'r', '0']
[102, 121, 111] ['B', 'o', '?', 'f', 'y', 'k', 'q', '`', '|', '#', 'o', 'j', 'n', 'c', 'z', 'q', 'r', '1']
[102, 121, 112] ['B', 'o', ' ', 'f', 'y', 't', 'q', '`', 'c', '#', 'o', 'u', 'n', 'c', 'e', 'q', 'r', '.']
[102, 121, 113] ['B', 'o', '!',

[103, 120, 107] ['C', 'n', ';', 'g', 'x', 'o', 'p', 'a', 'x', '"', 'n', 'n', 'o', 'b', '~', 'p', 's', '5']
[103, 120, 108] ['C', 'n', '<', 'g', 'x', 'h', 'p', 'a', '\x7f', '"', 'n', 'i', 'o', 'b', 'y', 'p', 's', '2']
[103, 120, 109] ['C', 'n', '=', 'g', 'x', 'i', 'p', 'a', '~', '"', 'n', 'h', 'o', 'b', 'x', 'p', 's', '3']
[103, 120, 110] ['C', 'n', '>', 'g', 'x', 'j', 'p', 'a', '}', '"', 'n', 'k', 'o', 'b', '{', 'p', 's', '0']
[103, 120, 111] ['C', 'n', '?', 'g', 'x', 'k', 'p', 'a', '|', '"', 'n', 'j', 'o', 'b', 'z', 'p', 's', '1']
[103, 120, 112] ['C', 'n', ' ', 'g', 'x', 't', 'p', 'a', 'c', '"', 'n', 'u', 'o', 'b', 'e', 'p', 's', '.']
[103, 120, 113] ['C', 'n', '!', 'g', 'x', 'u', 'p', 'a', 'b', '"', 'n', 't', 'o', 'b', 'd', 'p', 's', '/']
[103, 120, 114] ['C', 'n', '"', 'g', 'x', 'v', 'p', 'a', 'a', '"', 'n', 'w', 'o', 'b', 'g', 'p', 's', ',']
[103, 120, 115] ['C', 'n', '#', 'g', 'x', 'w', 'p', 'a', '`', '"', 'n', 'v', 'o', 'b', 'f', 'p', 's', '-']
[103, 120, 116] ['C', 'n', '$', 'g

[104, 113, 120] ['L', 'g', '(', 'h', 'q', '|', '\x7f', 'h', 'k', '-', 'g', '}', '`', 'k', 'm', '\x7f', 'z', '&']
[104, 113, 121] ['L', 'g', ')', 'h', 'q', '}', '\x7f', 'h', 'j', '-', 'g', '|', '`', 'k', 'l', '\x7f', 'z', "'"]
[104, 113, 122] ['L', 'g', '*', 'h', 'q', '~', '\x7f', 'h', 'i', '-', 'g', '\x7f', '`', 'k', 'o', '\x7f', 'z', '$']
[104, 114, 97] ['L', 'd', '1', 'h', 'r', 'e', '\x7f', 'k', 'r', '-', 'd', 'd', '`', 'h', 't', '\x7f', 'y', '?']
[104, 114, 98] ['L', 'd', '2', 'h', 'r', 'f', '\x7f', 'k', 'q', '-', 'd', 'g', '`', 'h', 'w', '\x7f', 'y', '<']
[104, 114, 99] ['L', 'd', '3', 'h', 'r', 'g', '\x7f', 'k', 'p', '-', 'd', 'f', '`', 'h', 'v', '\x7f', 'y', '=']
[104, 114, 100] ['L', 'd', '4', 'h', 'r', '`', '\x7f', 'k', 'w', '-', 'd', 'a', '`', 'h', 'q', '\x7f', 'y', ':']
[104, 114, 101] ['L', 'd', '5', 'h', 'r', 'a', '\x7f', 'k', 'v', '-', 'd', '`', '`', 'h', 'p', '\x7f', 'y', ';']
[104, 114, 102] ['L', 'd', '6', 'h', 'r', 'b', '\x7f', 'k', 'u', '-', 'd', 'c', '`', 'h', 's', '

[105, 110, 112] ['M', 'x', ' ', 'i', 'n', 't', '~', 'w', 'c', ',', 'x', 'u', 'a', 't', 'e', '~', 'e', '.']
[105, 110, 113] ['M', 'x', '!', 'i', 'n', 'u', '~', 'w', 'b', ',', 'x', 't', 'a', 't', 'd', '~', 'e', '/']
[105, 110, 114] ['M', 'x', '"', 'i', 'n', 'v', '~', 'w', 'a', ',', 'x', 'w', 'a', 't', 'g', '~', 'e', ',']
[105, 110, 115] ['M', 'x', '#', 'i', 'n', 'w', '~', 'w', '`', ',', 'x', 'v', 'a', 't', 'f', '~', 'e', '-']
[105, 110, 116] ['M', 'x', '$', 'i', 'n', 'p', '~', 'w', 'g', ',', 'x', 'q', 'a', 't', 'a', '~', 'e', '*']
[105, 110, 117] ['M', 'x', '%', 'i', 'n', 'q', '~', 'w', 'f', ',', 'x', 'p', 'a', 't', '`', '~', 'e', '+']
[105, 110, 118] ['M', 'x', '&', 'i', 'n', 'r', '~', 'w', 'e', ',', 'x', 's', 'a', 't', 'c', '~', 'e', '(']
[105, 110, 119] ['M', 'x', "'", 'i', 'n', 's', '~', 'w', 'd', ',', 'x', 'r', 'a', 't', 'b', '~', 'e', ')']
[105, 110, 120] ['M', 'x', '(', 'i', 'n', '|', '~', 'w', 'k', ',', 'x', '}', 'a', 't', 'm', '~', 'e', '&']
[105, 110, 121] ['M', 'x', ')', 'i', 

[106, 113, 107] ['N', 'g', ';', 'j', 'q', 'o', '}', 'h', 'x', '/', 'g', 'n', 'b', 'k', '~', '}', 'z', '5']
[106, 113, 108] ['N', 'g', '<', 'j', 'q', 'h', '}', 'h', '\x7f', '/', 'g', 'i', 'b', 'k', 'y', '}', 'z', '2']
[106, 113, 109] ['N', 'g', '=', 'j', 'q', 'i', '}', 'h', '~', '/', 'g', 'h', 'b', 'k', 'x', '}', 'z', '3']
[106, 113, 110] ['N', 'g', '>', 'j', 'q', 'j', '}', 'h', '}', '/', 'g', 'k', 'b', 'k', '{', '}', 'z', '0']
[106, 113, 111] ['N', 'g', '?', 'j', 'q', 'k', '}', 'h', '|', '/', 'g', 'j', 'b', 'k', 'z', '}', 'z', '1']
[106, 113, 112] ['N', 'g', ' ', 'j', 'q', 't', '}', 'h', 'c', '/', 'g', 'u', 'b', 'k', 'e', '}', 'z', '.']
[106, 113, 113] ['N', 'g', '!', 'j', 'q', 'u', '}', 'h', 'b', '/', 'g', 't', 'b', 'k', 'd', '}', 'z', '/']
[106, 113, 114] ['N', 'g', '"', 'j', 'q', 'v', '}', 'h', 'a', '/', 'g', 'w', 'b', 'k', 'g', '}', 'z', ',']
[106, 113, 115] ['N', 'g', '#', 'j', 'q', 'w', '}', 'h', '`', '/', 'g', 'v', 'b', 'k', 'f', '}', 'z', '-']
[106, 113, 116] ['N', 'g', '$', 'j

[107, 116, 103] ['O', 'b', '7', 'k', 't', 'c', '|', 'm', 't', '.', 'b', 'b', 'c', 'n', 'r', '|', '\x7f', '9']
[107, 116, 104] ['O', 'b', '8', 'k', 't', 'l', '|', 'm', '{', '.', 'b', 'm', 'c', 'n', '}', '|', '\x7f', '6']
[107, 116, 105] ['O', 'b', '9', 'k', 't', 'm', '|', 'm', 'z', '.', 'b', 'l', 'c', 'n', '|', '|', '\x7f', '7']
[107, 116, 106] ['O', 'b', ':', 'k', 't', 'n', '|', 'm', 'y', '.', 'b', 'o', 'c', 'n', '\x7f', '|', '\x7f', '4']
[107, 116, 107] ['O', 'b', ';', 'k', 't', 'o', '|', 'm', 'x', '.', 'b', 'n', 'c', 'n', '~', '|', '\x7f', '5']
[107, 116, 108] ['O', 'b', '<', 'k', 't', 'h', '|', 'm', '\x7f', '.', 'b', 'i', 'c', 'n', 'y', '|', '\x7f', '2']
[107, 116, 109] ['O', 'b', '=', 'k', 't', 'i', '|', 'm', '~', '.', 'b', 'h', 'c', 'n', 'x', '|', '\x7f', '3']
[107, 116, 110] ['O', 'b', '>', 'k', 't', 'j', '|', 'm', '}', '.', 'b', 'k', 'c', 'n', '{', '|', '\x7f', '0']
[107, 116, 111] ['O', 'b', '?', 'k', 't', 'k', '|', 'm', '|', '.', 'b', 'j', 'c', 'n', 'z', '|', '\x7f', '1']
[107

[108, 117, 102] ['H', 'c', '6', 'l', 'u', 'b', '{', 'l', 'u', ')', 'c', 'c', 'd', 'o', 's', '{', '~', '8']
[108, 117, 103] ['H', 'c', '7', 'l', 'u', 'c', '{', 'l', 't', ')', 'c', 'b', 'd', 'o', 'r', '{', '~', '9']
[108, 117, 104] ['H', 'c', '8', 'l', 'u', 'l', '{', 'l', '{', ')', 'c', 'm', 'd', 'o', '}', '{', '~', '6']
[108, 117, 105] ['H', 'c', '9', 'l', 'u', 'm', '{', 'l', 'z', ')', 'c', 'l', 'd', 'o', '|', '{', '~', '7']
[108, 117, 106] ['H', 'c', ':', 'l', 'u', 'n', '{', 'l', 'y', ')', 'c', 'o', 'd', 'o', '\x7f', '{', '~', '4']
[108, 117, 107] ['H', 'c', ';', 'l', 'u', 'o', '{', 'l', 'x', ')', 'c', 'n', 'd', 'o', '~', '{', '~', '5']
[108, 117, 108] ['H', 'c', '<', 'l', 'u', 'h', '{', 'l', '\x7f', ')', 'c', 'i', 'd', 'o', 'y', '{', '~', '2']
[108, 117, 109] ['H', 'c', '=', 'l', 'u', 'i', '{', 'l', '~', ')', 'c', 'h', 'd', 'o', 'x', '{', '~', '3']
[108, 117, 110] ['H', 'c', '>', 'l', 'u', 'j', '{', 'l', '}', ')', 'c', 'k', 'd', 'o', '{', '{', '~', '0']
[108, 117, 111] ['H', 'c', '?',

[109, 114, 112] ['I', 'd', ' ', 'm', 'r', 't', 'z', 'k', 'c', '(', 'd', 'u', 'e', 'h', 'e', 'z', 'y', '.']
[109, 114, 113] ['I', 'd', '!', 'm', 'r', 'u', 'z', 'k', 'b', '(', 'd', 't', 'e', 'h', 'd', 'z', 'y', '/']
[109, 114, 114] ['I', 'd', '"', 'm', 'r', 'v', 'z', 'k', 'a', '(', 'd', 'w', 'e', 'h', 'g', 'z', 'y', ',']
[109, 114, 115] ['I', 'd', '#', 'm', 'r', 'w', 'z', 'k', '`', '(', 'd', 'v', 'e', 'h', 'f', 'z', 'y', '-']
[109, 114, 116] ['I', 'd', '$', 'm', 'r', 'p', 'z', 'k', 'g', '(', 'd', 'q', 'e', 'h', 'a', 'z', 'y', '*']
[109, 114, 117] ['I', 'd', '%', 'm', 'r', 'q', 'z', 'k', 'f', '(', 'd', 'p', 'e', 'h', '`', 'z', 'y', '+']
[109, 114, 118] ['I', 'd', '&', 'm', 'r', 'r', 'z', 'k', 'e', '(', 'd', 's', 'e', 'h', 'c', 'z', 'y', '(']
[109, 114, 119] ['I', 'd', "'", 'm', 'r', 's', 'z', 'k', 'd', '(', 'd', 'r', 'e', 'h', 'b', 'z', 'y', ')']
[109, 114, 120] ['I', 'd', '(', 'm', 'r', '|', 'z', 'k', 'k', '(', 'd', '}', 'e', 'h', 'm', 'z', 'y', '&']
[109, 114, 121] ['I', 'd', ')', 'm', 

[110, 115, 100] ['J', 'e', '4', 'n', 's', '`', 'y', 'j', 'w', '+', 'e', 'a', 'f', 'i', 'q', 'y', 'x', ':']
[110, 115, 101] ['J', 'e', '5', 'n', 's', 'a', 'y', 'j', 'v', '+', 'e', '`', 'f', 'i', 'p', 'y', 'x', ';']
[110, 115, 102] ['J', 'e', '6', 'n', 's', 'b', 'y', 'j', 'u', '+', 'e', 'c', 'f', 'i', 's', 'y', 'x', '8']
[110, 115, 103] ['J', 'e', '7', 'n', 's', 'c', 'y', 'j', 't', '+', 'e', 'b', 'f', 'i', 'r', 'y', 'x', '9']
[110, 115, 104] ['J', 'e', '8', 'n', 's', 'l', 'y', 'j', '{', '+', 'e', 'm', 'f', 'i', '}', 'y', 'x', '6']
[110, 115, 105] ['J', 'e', '9', 'n', 's', 'm', 'y', 'j', 'z', '+', 'e', 'l', 'f', 'i', '|', 'y', 'x', '7']
[110, 115, 106] ['J', 'e', ':', 'n', 's', 'n', 'y', 'j', 'y', '+', 'e', 'o', 'f', 'i', '\x7f', 'y', 'x', '4']
[110, 115, 107] ['J', 'e', ';', 'n', 's', 'o', 'y', 'j', 'x', '+', 'e', 'n', 'f', 'i', '~', 'y', 'x', '5']
[110, 115, 108] ['J', 'e', '<', 'n', 's', 'h', 'y', 'j', '\x7f', '+', 'e', 'i', 'f', 'i', 'y', 'y', 'x', '2']
[110, 115, 109] ['J', 'e', '=',

[111, 117, 122] ['K', 'c', '*', 'o', 'u', '~', 'x', 'l', 'i', '*', 'c', '\x7f', 'g', 'o', 'o', 'x', '~', '$']
[111, 118, 97] ['K', '`', '1', 'o', 'v', 'e', 'x', 'o', 'r', '*', '`', 'd', 'g', 'l', 't', 'x', '}', '?']
[111, 118, 98] ['K', '`', '2', 'o', 'v', 'f', 'x', 'o', 'q', '*', '`', 'g', 'g', 'l', 'w', 'x', '}', '<']
[111, 118, 99] ['K', '`', '3', 'o', 'v', 'g', 'x', 'o', 'p', '*', '`', 'f', 'g', 'l', 'v', 'x', '}', '=']
[111, 118, 100] ['K', '`', '4', 'o', 'v', '`', 'x', 'o', 'w', '*', '`', 'a', 'g', 'l', 'q', 'x', '}', ':']
[111, 118, 101] ['K', '`', '5', 'o', 'v', 'a', 'x', 'o', 'v', '*', '`', '`', 'g', 'l', 'p', 'x', '}', ';']
[111, 118, 102] ['K', '`', '6', 'o', 'v', 'b', 'x', 'o', 'u', '*', '`', 'c', 'g', 'l', 's', 'x', '}', '8']
[111, 118, 103] ['K', '`', '7', 'o', 'v', 'c', 'x', 'o', 't', '*', '`', 'b', 'g', 'l', 'r', 'x', '}', '9']
[111, 118, 104] ['K', '`', '8', 'o', 'v', 'l', 'x', 'o', '{', '*', '`', 'm', 'g', 'l', '}', 'x', '}', '6']
[111, 118, 105] ['K', '`', '9', 'o', 

[112, 115, 104] ['T', 'e', '8', 'p', 's', 'l', 'g', 'j', '{', '5', 'e', 'm', 'x', 'i', '}', 'g', 'x', '6']
[112, 115, 105] ['T', 'e', '9', 'p', 's', 'm', 'g', 'j', 'z', '5', 'e', 'l', 'x', 'i', '|', 'g', 'x', '7']
[112, 115, 106] ['T', 'e', ':', 'p', 's', 'n', 'g', 'j', 'y', '5', 'e', 'o', 'x', 'i', '\x7f', 'g', 'x', '4']
[112, 115, 107] ['T', 'e', ';', 'p', 's', 'o', 'g', 'j', 'x', '5', 'e', 'n', 'x', 'i', '~', 'g', 'x', '5']
[112, 115, 108] ['T', 'e', '<', 'p', 's', 'h', 'g', 'j', '\x7f', '5', 'e', 'i', 'x', 'i', 'y', 'g', 'x', '2']
[112, 115, 109] ['T', 'e', '=', 'p', 's', 'i', 'g', 'j', '~', '5', 'e', 'h', 'x', 'i', 'x', 'g', 'x', '3']
[112, 115, 110] ['T', 'e', '>', 'p', 's', 'j', 'g', 'j', '}', '5', 'e', 'k', 'x', 'i', '{', 'g', 'x', '0']
[112, 115, 111] ['T', 'e', '?', 'p', 's', 'k', 'g', 'j', '|', '5', 'e', 'j', 'x', 'i', 'z', 'g', 'x', '1']
[112, 115, 112] ['T', 'e', ' ', 'p', 's', 't', 'g', 'j', 'c', '5', 'e', 'u', 'x', 'i', 'e', 'g', 'x', '.']
[112, 115, 113] ['T', 'e', '!',

[113, 112, 117] ['U', 'f', '%', 'q', 'p', 'q', 'f', 'i', 'f', '4', 'f', 'p', 'y', 'j', '`', 'f', '{', '+']
[113, 112, 118] ['U', 'f', '&', 'q', 'p', 'r', 'f', 'i', 'e', '4', 'f', 's', 'y', 'j', 'c', 'f', '{', '(']
[113, 112, 119] ['U', 'f', "'", 'q', 'p', 's', 'f', 'i', 'd', '4', 'f', 'r', 'y', 'j', 'b', 'f', '{', ')']
[113, 112, 120] ['U', 'f', '(', 'q', 'p', '|', 'f', 'i', 'k', '4', 'f', '}', 'y', 'j', 'm', 'f', '{', '&']
[113, 112, 121] ['U', 'f', ')', 'q', 'p', '}', 'f', 'i', 'j', '4', 'f', '|', 'y', 'j', 'l', 'f', '{', "'"]
[113, 112, 122] ['U', 'f', '*', 'q', 'p', '~', 'f', 'i', 'i', '4', 'f', '\x7f', 'y', 'j', 'o', 'f', '{', '$']
[113, 113, 97] ['U', 'g', '1', 'q', 'q', 'e', 'f', 'h', 'r', '4', 'g', 'd', 'y', 'k', 't', 'f', 'z', '?']
[113, 113, 98] ['U', 'g', '2', 'q', 'q', 'f', 'f', 'h', 'q', '4', 'g', 'g', 'y', 'k', 'w', 'f', 'z', '<']
[113, 113, 99] ['U', 'g', '3', 'q', 'q', 'g', 'f', 'h', 'p', '4', 'g', 'f', 'y', 'k', 'v', 'f', 'z', '=']
[113, 113, 100] ['U', 'g', '4', 'q', 

[114, 107, 103] ['V', '}', '7', 'r', 'k', 'c', 'e', 'r', 't', '7', '}', 'b', 'z', 'q', 'r', 'e', '`', '9']
[114, 107, 104] ['V', '}', '8', 'r', 'k', 'l', 'e', 'r', '{', '7', '}', 'm', 'z', 'q', '}', 'e', '`', '6']
[114, 107, 105] ['V', '}', '9', 'r', 'k', 'm', 'e', 'r', 'z', '7', '}', 'l', 'z', 'q', '|', 'e', '`', '7']
[114, 107, 106] ['V', '}', ':', 'r', 'k', 'n', 'e', 'r', 'y', '7', '}', 'o', 'z', 'q', '\x7f', 'e', '`', '4']
[114, 107, 107] ['V', '}', ';', 'r', 'k', 'o', 'e', 'r', 'x', '7', '}', 'n', 'z', 'q', '~', 'e', '`', '5']
[114, 107, 108] ['V', '}', '<', 'r', 'k', 'h', 'e', 'r', '\x7f', '7', '}', 'i', 'z', 'q', 'y', 'e', '`', '2']
[114, 107, 109] ['V', '}', '=', 'r', 'k', 'i', 'e', 'r', '~', '7', '}', 'h', 'z', 'q', 'x', 'e', '`', '3']
[114, 107, 110] ['V', '}', '>', 'r', 'k', 'j', 'e', 'r', '}', '7', '}', 'k', 'z', 'q', '{', 'e', '`', '0']
[114, 107, 111] ['V', '}', '?', 'r', 'k', 'k', 'e', 'r', '|', '7', '}', 'j', 'z', 'q', 'z', 'e', '`', '1']
[114, 107, 112] ['V', '}', ' ',

[115, 103, 98] ['W', 'q', '2', 's', 'g', 'f', 'd', '~', 'q', '6', 'q', 'g', '{', '}', 'w', 'd', 'l', '<']
[115, 103, 99] ['W', 'q', '3', 's', 'g', 'g', 'd', '~', 'p', '6', 'q', 'f', '{', '}', 'v', 'd', 'l', '=']
[115, 103, 100] ['W', 'q', '4', 's', 'g', '`', 'd', '~', 'w', '6', 'q', 'a', '{', '}', 'q', 'd', 'l', ':']
[115, 103, 101] ['W', 'q', '5', 's', 'g', 'a', 'd', '~', 'v', '6', 'q', '`', '{', '}', 'p', 'd', 'l', ';']
[115, 103, 102] ['W', 'q', '6', 's', 'g', 'b', 'd', '~', 'u', '6', 'q', 'c', '{', '}', 's', 'd', 'l', '8']
[115, 103, 103] ['W', 'q', '7', 's', 'g', 'c', 'd', '~', 't', '6', 'q', 'b', '{', '}', 'r', 'd', 'l', '9']
[115, 103, 104] ['W', 'q', '8', 's', 'g', 'l', 'd', '~', '{', '6', 'q', 'm', '{', '}', '}', 'd', 'l', '6']
[115, 103, 105] ['W', 'q', '9', 's', 'g', 'm', 'd', '~', 'z', '6', 'q', 'l', '{', '}', '|', 'd', 'l', '7']
[115, 103, 106] ['W', 'q', ':', 's', 'g', 'n', 'd', '~', 'y', '6', 'q', 'o', '{', '}', '\x7f', 'd', 'l', '4']
[115, 103, 107] ['W', 'q', ';', 's',

[116, 101, 104] ['P', 's', '8', 't', 'e', 'l', 'c', '|', '{', '1', 's', 'm', '|', '\x7f', '}', 'c', 'n', '6']
[116, 101, 105] ['P', 's', '9', 't', 'e', 'm', 'c', '|', 'z', '1', 's', 'l', '|', '\x7f', '|', 'c', 'n', '7']
[116, 101, 106] ['P', 's', ':', 't', 'e', 'n', 'c', '|', 'y', '1', 's', 'o', '|', '\x7f', '\x7f', 'c', 'n', '4']
[116, 101, 107] ['P', 's', ';', 't', 'e', 'o', 'c', '|', 'x', '1', 's', 'n', '|', '\x7f', '~', 'c', 'n', '5']
[116, 101, 108] ['P', 's', '<', 't', 'e', 'h', 'c', '|', '\x7f', '1', 's', 'i', '|', '\x7f', 'y', 'c', 'n', '2']
[116, 101, 109] ['P', 's', '=', 't', 'e', 'i', 'c', '|', '~', '1', 's', 'h', '|', '\x7f', 'x', 'c', 'n', '3']
[116, 101, 110] ['P', 's', '>', 't', 'e', 'j', 'c', '|', '}', '1', 's', 'k', '|', '\x7f', '{', 'c', 'n', '0']
[116, 101, 111] ['P', 's', '?', 't', 'e', 'k', 'c', '|', '|', '1', 's', 'j', '|', '\x7f', 'z', 'c', 'n', '1']
[116, 101, 112] ['P', 's', ' ', 't', 'e', 't', 'c', '|', 'c', '1', 's', 'u', '|', '\x7f', 'e', 'c', 'n', '.']
[116

[117, 101, 97] ['Q', 's', '1', 'u', 'e', 'e', 'b', '|', 'r', '0', 's', 'd', '}', '\x7f', 't', 'b', 'n', '?']
[117, 101, 98] ['Q', 's', '2', 'u', 'e', 'f', 'b', '|', 'q', '0', 's', 'g', '}', '\x7f', 'w', 'b', 'n', '<']
[117, 101, 99] ['Q', 's', '3', 'u', 'e', 'g', 'b', '|', 'p', '0', 's', 'f', '}', '\x7f', 'v', 'b', 'n', '=']
[117, 101, 100] ['Q', 's', '4', 'u', 'e', '`', 'b', '|', 'w', '0', 's', 'a', '}', '\x7f', 'q', 'b', 'n', ':']
[117, 101, 101] ['Q', 's', '5', 'u', 'e', 'a', 'b', '|', 'v', '0', 's', '`', '}', '\x7f', 'p', 'b', 'n', ';']
[117, 101, 102] ['Q', 's', '6', 'u', 'e', 'b', 'b', '|', 'u', '0', 's', 'c', '}', '\x7f', 's', 'b', 'n', '8']
[117, 101, 103] ['Q', 's', '7', 'u', 'e', 'c', 'b', '|', 't', '0', 's', 'b', '}', '\x7f', 'r', 'b', 'n', '9']
[117, 101, 104] ['Q', 's', '8', 'u', 'e', 'l', 'b', '|', '{', '0', 's', 'm', '}', '\x7f', '}', 'b', 'n', '6']
[117, 101, 105] ['Q', 's', '9', 'u', 'e', 'm', 'b', '|', 'z', '0', 's', 'l', '}', '\x7f', '|', 'b', 'n', '7']
[117, 101, 10

[118, 99, 106] ['R', 'u', ':', 'v', 'c', 'n', 'a', 'z', 'y', '3', 'u', 'o', '~', 'y', '\x7f', 'a', 'h', '4']
[118, 99, 107] ['R', 'u', ';', 'v', 'c', 'o', 'a', 'z', 'x', '3', 'u', 'n', '~', 'y', '~', 'a', 'h', '5']
[118, 99, 108] ['R', 'u', '<', 'v', 'c', 'h', 'a', 'z', '\x7f', '3', 'u', 'i', '~', 'y', 'y', 'a', 'h', '2']
[118, 99, 109] ['R', 'u', '=', 'v', 'c', 'i', 'a', 'z', '~', '3', 'u', 'h', '~', 'y', 'x', 'a', 'h', '3']
[118, 99, 110] ['R', 'u', '>', 'v', 'c', 'j', 'a', 'z', '}', '3', 'u', 'k', '~', 'y', '{', 'a', 'h', '0']
[118, 99, 111] ['R', 'u', '?', 'v', 'c', 'k', 'a', 'z', '|', '3', 'u', 'j', '~', 'y', 'z', 'a', 'h', '1']
[118, 99, 112] ['R', 'u', ' ', 'v', 'c', 't', 'a', 'z', 'c', '3', 'u', 'u', '~', 'y', 'e', 'a', 'h', '.']
[118, 99, 113] ['R', 'u', '!', 'v', 'c', 'u', 'a', 'z', 'b', '3', 'u', 't', '~', 'y', 'd', 'a', 'h', '/']
[118, 99, 114] ['R', 'u', '"', 'v', 'c', 'v', 'a', 'z', 'a', '3', 'u', 'w', '~', 'y', 'g', 'a', 'h', ',']
[118, 99, 115] ['R', 'u', '#', 'v', 'c',

[118, 122, 114] ['R', 'l', '"', 'v', 'z', 'v', 'a', 'c', 'a', '3', 'l', 'w', '~', '`', 'g', 'a', 'q', ',']
[118, 122, 115] ['R', 'l', '#', 'v', 'z', 'w', 'a', 'c', '`', '3', 'l', 'v', '~', '`', 'f', 'a', 'q', '-']
[118, 122, 116] ['R', 'l', '$', 'v', 'z', 'p', 'a', 'c', 'g', '3', 'l', 'q', '~', '`', 'a', 'a', 'q', '*']
[118, 122, 117] ['R', 'l', '%', 'v', 'z', 'q', 'a', 'c', 'f', '3', 'l', 'p', '~', '`', '`', 'a', 'q', '+']
[118, 122, 118] ['R', 'l', '&', 'v', 'z', 'r', 'a', 'c', 'e', '3', 'l', 's', '~', '`', 'c', 'a', 'q', '(']
[118, 122, 119] ['R', 'l', "'", 'v', 'z', 's', 'a', 'c', 'd', '3', 'l', 'r', '~', '`', 'b', 'a', 'q', ')']
[118, 122, 120] ['R', 'l', '(', 'v', 'z', '|', 'a', 'c', 'k', '3', 'l', '}', '~', '`', 'm', 'a', 'q', '&']
[118, 122, 121] ['R', 'l', ')', 'v', 'z', '}', 'a', 'c', 'j', '3', 'l', '|', '~', '`', 'l', 'a', 'q', "'"]
[118, 122, 122] ['R', 'l', '*', 'v', 'z', '~', 'a', 'c', 'i', '3', 'l', '\x7f', '~', '`', 'o', 'a', 'q', '$']
[119, 97, 97] ['S', 'w', '1', 'w',

[119, 112, 100] ['S', 'f', '4', 'w', 'p', '`', '`', 'i', 'w', '2', 'f', 'a', '\x7f', 'j', 'q', '`', '{', ':']
[119, 112, 101] ['S', 'f', '5', 'w', 'p', 'a', '`', 'i', 'v', '2', 'f', '`', '\x7f', 'j', 'p', '`', '{', ';']
[119, 112, 102] ['S', 'f', '6', 'w', 'p', 'b', '`', 'i', 'u', '2', 'f', 'c', '\x7f', 'j', 's', '`', '{', '8']
[119, 112, 103] ['S', 'f', '7', 'w', 'p', 'c', '`', 'i', 't', '2', 'f', 'b', '\x7f', 'j', 'r', '`', '{', '9']
[119, 112, 104] ['S', 'f', '8', 'w', 'p', 'l', '`', 'i', '{', '2', 'f', 'm', '\x7f', 'j', '}', '`', '{', '6']
[119, 112, 105] ['S', 'f', '9', 'w', 'p', 'm', '`', 'i', 'z', '2', 'f', 'l', '\x7f', 'j', '|', '`', '{', '7']
[119, 112, 106] ['S', 'f', ':', 'w', 'p', 'n', '`', 'i', 'y', '2', 'f', 'o', '\x7f', 'j', '\x7f', '`', '{', '4']
[119, 112, 107] ['S', 'f', ';', 'w', 'p', 'o', '`', 'i', 'x', '2', 'f', 'n', '\x7f', 'j', '~', '`', '{', '5']
[119, 112, 108] ['S', 'f', '<', 'w', 'p', 'h', '`', 'i', '\x7f', '2', 'f', 'i', '\x7f', 'j', 'y', '`', '{', '2']
[119

In [30]:
## testing
sample_message = ASCII_int_list[0:9] + ASCII_int_list[-9:]

for first in range(97, 122 + 1):
    for third in [112]:
        for second in range(97, 122 + 1):
#             second = 97
        
#             third = 97
    
            decoded_message = []
    
            decoder = [first, second, third]
            
            for idx in range(len(sample_message)):
                
                decoded_ASCII = sample_message[idx] ^ decoder[idx%3]
                
                decoded_message.append(decoded_ASCII)
                
                
            is_valid=True
            
            if (decoded_message[0] >= 65 and decoded_message[0] <= 90) or (decoded_message[0] >= 97 and decoded_message[0] <= 122):
                decoded_message = [chr(ASCII) for ASCII in decoded_message]
            
                print(decoder, decoded_message)

[97, 97, 112] ['E', 'w', ' ', 'a', 'a', 't', 'v', 'x', 'c', '$', 'w', 'u', 'i', '{', 'e', 'v', 'j', '.']
[97, 98, 112] ['E', 't', ' ', 'a', 'b', 't', 'v', '{', 'c', '$', 't', 'u', 'i', 'x', 'e', 'v', 'i', '.']
[97, 99, 112] ['E', 'u', ' ', 'a', 'c', 't', 'v', 'z', 'c', '$', 'u', 'u', 'i', 'y', 'e', 'v', 'h', '.']
[97, 100, 112] ['E', 'r', ' ', 'a', 'd', 't', 'v', '}', 'c', '$', 'r', 'u', 'i', '~', 'e', 'v', 'o', '.']
[97, 101, 112] ['E', 's', ' ', 'a', 'e', 't', 'v', '|', 'c', '$', 's', 'u', 'i', '\x7f', 'e', 'v', 'n', '.']
[97, 102, 112] ['E', 'p', ' ', 'a', 'f', 't', 'v', '\x7f', 'c', '$', 'p', 'u', 'i', '|', 'e', 'v', 'm', '.']
[97, 103, 112] ['E', 'q', ' ', 'a', 'g', 't', 'v', '~', 'c', '$', 'q', 'u', 'i', '}', 'e', 'v', 'l', '.']
[97, 104, 112] ['E', '~', ' ', 'a', 'h', 't', 'v', 'q', 'c', '$', '~', 'u', 'i', 'r', 'e', 'v', 'c', '.']
[97, 105, 112] ['E', '\x7f', ' ', 'a', 'i', 't', 'v', 'p', 'c', '$', '\x7f', 'u', 'i', 's', 'e', 'v', 'b', '.']
[97, 106, 112] ['E', '|', ' ', 'a', '

In [33]:
## testing
sample_message = ASCII_int_list[0:9] + ASCII_int_list[-9:]

for first in range(97, 122 + 1):
    for third in [112]:
        for second in [99, 108, 110, 111, 113, 114, 120]:
#             second = 97
        
#             third = 97
    
            decoded_message = []
    
            decoder = [first, second, third]
            
            for idx in range(len(sample_message)):
                
                decoded_ASCII = sample_message[idx] ^ decoder[idx%3]
                
                decoded_message.append(decoded_ASCII)
                
                
            is_valid=True
            
            if (decoded_message[0] >= 65 and decoded_message[0] <= 90) or (decoded_message[0] >= 97 and decoded_message[0] <= 122):
                decoded_message = [chr(ASCII) for ASCII in decoded_message]
            
                print(decoder, decoded_message)

[97, 99, 112] ['E', 'u', ' ', 'a', 'c', 't', 'v', 'z', 'c', '$', 'u', 'u', 'i', 'y', 'e', 'v', 'h', '.']
[97, 108, 112] ['E', 'z', ' ', 'a', 'l', 't', 'v', 'u', 'c', '$', 'z', 'u', 'i', 'v', 'e', 'v', 'g', '.']
[97, 110, 112] ['E', 'x', ' ', 'a', 'n', 't', 'v', 'w', 'c', '$', 'x', 'u', 'i', 't', 'e', 'v', 'e', '.']
[97, 111, 112] ['E', 'y', ' ', 'a', 'o', 't', 'v', 'v', 'c', '$', 'y', 'u', 'i', 'u', 'e', 'v', 'd', '.']
[97, 113, 112] ['E', 'g', ' ', 'a', 'q', 't', 'v', 'h', 'c', '$', 'g', 'u', 'i', 'k', 'e', 'v', 'z', '.']
[97, 114, 112] ['E', 'd', ' ', 'a', 'r', 't', 'v', 'k', 'c', '$', 'd', 'u', 'i', 'h', 'e', 'v', 'y', '.']
[97, 120, 112] ['E', 'n', ' ', 'a', 'x', 't', 'v', 'a', 'c', '$', 'n', 'u', 'i', 'b', 'e', 'v', 's', '.']
[98, 99, 112] ['F', 'u', ' ', 'b', 'c', 't', 'u', 'z', 'c', "'", 'u', 'u', 'j', 'y', 'e', 'u', 'h', '.']
[98, 108, 112] ['F', 'z', ' ', 'b', 'l', 't', 'u', 'u', 'c', "'", 'z', 'u', 'j', 'v', 'e', 'u', 'g', '.']
[98, 110, 112] ['F', 'x', ' ', 'b', 'n', 't', 'u

In [37]:
## testing
sample_message = ASCII_int_list[0:18]

for first in [101, 103, 105, 107, 108, 109]:
    for third in [112]:
        for second in [99, 108, 110, 111, 113, 114, 120]:
#             second = 97
        
#             third = 97
    
            decoded_message = []
    
            decoder = [first, second, third]
            
            for idx in range(len(sample_message)):
                
                decoded_ASCII = sample_message[idx] ^ decoder[idx%3]
                
                decoded_message.append(decoded_ASCII)
                
                
            is_valid=True
            
            if (decoded_message[0] >= 65 and decoded_message[0] <= 90) or (decoded_message[0] >= 97 and decoded_message[0] <= 122):
                decoded_message = [chr(ASCII) for ASCII in decoded_message]
            
                print(decoder, decoded_message)

[101, 99, 112] ['A', 'u', ' ', 'e', 'c', 't', 'r', 'z', 'c', 't', ';', 't', 'a', 'p', 'e', 'n', ';', 'f']
[101, 108, 112] ['A', 'z', ' ', 'e', 'l', 't', 'r', 'u', 'c', 't', '4', 't', 'a', '\x7f', 'e', 'n', '4', 'f']
[101, 110, 112] ['A', 'x', ' ', 'e', 'n', 't', 'r', 'w', 'c', 't', '6', 't', 'a', '}', 'e', 'n', '6', 'f']
[101, 111, 112] ['A', 'y', ' ', 'e', 'o', 't', 'r', 'v', 'c', 't', '7', 't', 'a', '|', 'e', 'n', '7', 'f']
[101, 113, 112] ['A', 'g', ' ', 'e', 'q', 't', 'r', 'h', 'c', 't', ')', 't', 'a', 'b', 'e', 'n', ')', 'f']
[101, 114, 112] ['A', 'd', ' ', 'e', 'r', 't', 'r', 'k', 'c', 't', '*', 't', 'a', 'a', 'e', 'n', '*', 'f']
[101, 120, 112] ['A', 'n', ' ', 'e', 'x', 't', 'r', 'a', 'c', 't', ' ', 't', 'a', 'k', 'e', 'n', ' ', 'f']
[103, 99, 112] ['C', 'u', ' ', 'g', 'c', 't', 'p', 'z', 'c', 'v', ';', 't', 'c', 'p', 'e', 'l', ';', 'f']
[103, 108, 112] ['C', 'z', ' ', 'g', 'l', 't', 'p', 'u', 'c', 'v', '4', 't', 'c', '\x7f', 'e', 'l', '4', 'f']
[103, 110, 112] ['C', 'x', ' ', '

In [39]:
## found it !
## testing
sample_message = ASCII_int_list

first = 101
second = 120
third = 112

decoded_message = []

decoder = [first, second, third]

for idx in range(len(sample_message)):

    decoded_ASCII = sample_message[idx] ^ decoder[idx%3]

    decoded_message.append(decoded_ASCII)


is_valid=True

if (decoded_message[0] >= 65 and decoded_message[0] <= 90) or (decoded_message[0] >= 97 and decoded_message[0] <= 122):
    decoded_message = [chr(ASCII) for ASCII in decoded_message]

    print(decoder, decoded_message)

[101, 120, 112] ['A', 'n', ' ', 'e', 'x', 't', 'r', 'a', 'c', 't', ' ', 't', 'a', 'k', 'e', 'n', ' ', 'f', 'r', 'o', 'm', ' ', 't', 'h', 'e', ' ', 'i', 'n', 't', 'r', 'o', 'd', 'u', 'c', 't', 'i', 'o', 'n', ' ', 'o', 'f', ' ', 'o', 'n', 'e', ' ', 'o', 'f', ' ', 'E', 'u', 'l', 'e', 'r', "'", 's', ' ', 'm', 'o', 's', 't', ' ', 'c', 'e', 'l', 'e', 'b', 'r', 'a', 't', 'e', 'd', ' ', 'p', 'a', 'p', 'e', 'r', 's', ',', ' ', '"', 'D', 'e', ' ', 's', 'u', 'm', 'm', 'i', 's', ' ', 's', 'e', 'r', 'i', 'e', 'r', 'u', 'm', ' ', 'r', 'e', 'c', 'i', 'p', 'r', 'o', 'c', 'a', 'r', 'u', 'm', '"', ' ', '[', 'O', 'n', ' ', 't', 'h', 'e', ' ', 's', 'u', 'm', 's', ' ', 'o', 'f', ' ', 's', 'e', 'r', 'i', 'e', 's', ' ', 'o', 'f', ' ', 'r', 'e', 'c', 'i', 'p', 'r', 'o', 'c', 'a', 'l', 's', ']', ':', ' ', 'I', ' ', 'h', 'a', 'v', 'e', ' ', 'r', 'e', 'c', 'e', 'n', 't', 'l', 'y', ' ', 'f', 'o', 'u', 'n', 'd', ',', ' ', 'q', 'u', 'i', 't', 'e', ' ', 'u', 'n', 'e', 'x', 'p', 'e', 'c', 't', 'e', 'd', 'l', 'y', ','

In [41]:
## solution
first = 101
second = 120
third = 112

decoder = [first, second, third]

decoded_ASCII_int_list = []

for idx in range(len(ASCII_int_list)):
    
    decoded_ASCII = ASCII_int_list[idx] ^ decoder[idx%3]
    
    decoded_ASCII_int_list.append(decoded_ASCII)
    
print(f"Answer is: {sum(decoded_ASCII_int_list)}")

Answer is: 129448


## Solution
The sums of the original ASCII int values is 129448

Time taken: from ~3 to 4:20 - around one and a half hours...

My approach was very haphazard and slow